doc/contributing/submitting-changes.chapter.md

```diff
@@ -234,6 +234,17 @@ digraph {
 
 The `master` branch is the main development branch. It should only see non-breaking commits that do not cause mass rebuilds.
 
+What is a mass rebuild? It's an update that takes Hydra build infrastructure long time to process (say, 8 hours).
+
+Here are rules of thumb that should help you decide if a change is reasonable for `master` branch:
+
+- Small update that rebuilds under 10 dependent packages unlikely to break users' systems. Examples are leaf package updates.
+- Important update with rebuild count less than 1000 dependent packages and does not contain heavyweight packages like `chromium` or `firefox`.
+- Kernel updates.
+- Urgent security fixes.
+
+Otherwise consider `staging` branch.
```

Comment:
I am not sure if we should mention that stdenv or core programming language rebuilds like python need to go to staging.

Reply:
I think it's reasonable to provide a few "clearly for staging" examples. Added as

"""
+Otherwise consider `staging` branch. Typical changes for `staging` are:
+
+- Any changes to `stdenv` (trigger full rebuld)
+- Any core language ecosystem changes like `python` or `ghc` update (usually trigger more than 1000 dependent packages rebuild)
"""